workflow-to-js-transformer/src/main/java/com/github/cafdataprocessing/workflow/transform/FullWorkflowRetriever.java

```diff
@@ -178,4 +240,61 @@ private FullAction buildFullAction(
         }
         return new FullAction(existingAction, actionConditions);
     }
+
+    private long getWorkflowId(final String projectId, final String workflowName)
+        throws ApiException, InvalidWorkflowSpecificationException
+    {
+        final WorkflowsApi workflowsApi = apisProvider.getWorkflowsApi();
+        final ExistingWorkflows existingWorkflows = workflowsApi.getWorkflows(projectId, 1, 100);
+        final Map<String, Long> workflows = existingWorkflows.getWorkflows().stream().collect(
+            Collectors.toMap(ExistingWorkflow::getName, ExistingWorkflow::getId));
```

Comment:
Good job on how you're constructing the map.

Reply:
That said, since you're not caching the map (i.e. you're only using it once), there's no real need to create it - you could do what you're trying to do more efficiently without a map.

Reply:
Thanks, I'll use this method of searching through the stream of objects in the future.